### PR TITLE
new plugin: build

### PR DESCRIPTION
**Pull request by kvaps**
New plugin kaniko.
Allows to build your dockerfiles directly in Kubernetes cluster.

examples:
```bash
# Show all kaniko commands
kubectl build --help

# Build from current directory
kubectl build --context . --no-push

# Specify namespace and kubeconfig
kubectl build --context . --no-push --namespace default --kubeconfig ~/.kube/someconfig

# Login to remote registry
docker login docker.io

# Short form
kubectl build -c . -d docker.io/some/image:latest

# Use cache building
kubectl build --context . --destination docker.io/some/image:latest --cache-repo docker.io/some/cache

# Build from stdin
tar -cvf- . | kubectl build --destination docker.io/some/image:latest --context tar://stdin
```

no special privileges needed

### CODE REVIEW

**Comment by ahmetb** [comment]
I suspect you aren’t the owner of the kaniko project. We don't allow third parties to reuse well-known project names that do not belong to them. If this was coming from the Kaniko project maintainers, we could accept. Otherwise, we have a no way of knowing you gave the blessing of Kaniko maintainers to use their name and portray your project as "official plugin" of Kaniko.  

/hold

**Comment by kvaps** [comment]
OK I guess `kubectl build` will also not be accepted, any other ideas? `kubectl kbuild`?

File: plugins/kaniko.yaml
Comment (corneliusweig) [inline plugins/kaniko.yaml:36]: The description is not the ideal place for your user docs. These examples are better suited for `kubectl kaniko --help`.

File: plugins/kaniko.yaml
Comment (corneliusweig) [inline plugins/kaniko.yaml:10]: Please state that you run a kaniko executor under the hood to build the images.

Also for those not familiar with kaniko you should briefly explain where the built image ends up.

**Review by corneliusweig — CHANGES_REQUESTED** [review]
Thanks you for another submission. This is certainly a nice-to-have plugin. However, it uses the name of a pretty well-known project, which is not ok.
There are two ways forward:

1) Rename the plugin to avoid the name clash
2) Work together with the kaniko folks and persuade them that the plugin is a good idea. If the script is hosted in their repo, the current name is perfect. Best you open an issue there and discuss this.

/hold

**Comment by corneliusweig** [comment]
Lol how could I overlook the previous conversation? I'll blame it on my patchy WiFi *grrr*  :wink:

File: plugins/kaniko.yaml
Comment (kvaps) [reply plugins/kaniko.yaml]: Yeah, amount of examples reduced.

File: plugins/kaniko.yaml
Comment (kvaps) [reply plugins/kaniko.yaml]: description updated!

**Comment by ahmetb** [comment]
Alternatively, you can currently [self-host an index](https://krew.sigs.k8s.io/docs/developer-guide/custom-indexes/) until you settle the naming usage rights. Your users will need to add your repo as the index and do `kubectl krew install kvaps/kaniko`. It really takes 1 additional command as far as the users are concerned.

**Comment by kvaps** [comment]
Yep, that's the way to go. @ahmetb have you plans to create common hub like Helm Hub / Operators Hub to concentrate all custom krew repositories in one place?

**Comment by ahmetb** [comment]
Not right now. If your concern is discoverability, once you get approvals from Kaniko maintainers or merge this into Kaniko project, you should be able to join this repository.

**Comment by kvaps** [comment]
Btw, I decided to rename **kubectl-kaniko** to **kubectl-build** and place it in my own krew-index repository.

**kubectl build** is better name because it do not conflicts with kaniko, and reflects exact usage, and also corelates with **docker build**:

```console
$ ls
Dockerfile
$ kubectl build -c . -d kvpas/myimage
```

The kubectl-build contradicts the plugin naming rules in krew-index, but I like it too much and do not intend to change.

@corneliusweig, many thanks you for review and suggestions. If you like this plugin, new you can install using my index repo:

```
kubectl krew index add kvaps https://github.com/kvaps/krew-index
kubectl krew install kvaps/build
```

**Comment by ahmetb** [comment]
cc: @chriskim06

**Pull request closed without merging**
